File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.c
int included(char * s, char c, int i, int j){
    for (int k = i; k<j; ++k){
        if (s[k] == c)
            return 1;
    }
    return 0;
}


int lengthOfLongestSubstring(char * s){
    int longestSubstr = 0;
    int j = 0;
    for (int i = 0; s[i] != '\0'; ++i){
        j = i+1;
        while (s[j] != '\0' && !included(s, s[j], i, j)){
            ++j;
        }
        if (j-i > longestSubstr)
            longestSubstr = j-i;
    }
    return longestSubstr;
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        i, j, n = 0, 1, len(s)
        longest = 0

        # range of unicode values for numbers, symbols and letters
        chars = [False for _ in range(128)]

        while (i < j <= n):
            # "use" char at position i
            chars[ord(s[i])] = True
            
            while j < n and not chars[ord(s[j])]:
                # find new unseen characters until repeated
                chars[ord(s[j])] = True
                j += 1
            
            longest = max(longest, j-i)
            if j == n:
                # longest substring was found
                break

            while s[i] != s[j]:
                # go up to one over the first occurence of the repeated character
                chars[ord(s[i])] = False
                i += 1

            i += 1
            j += 1
        
        return longest